src/app/project-notifications/project-notifications.component.ts

```diff
@@ -0,0 +1,303 @@
+import { TableParamsObject } from 'app/shared/components/table-template/table-params-object';
+import { TableTemplateUtils } from 'app/shared/utils/table-template-utils';
+import { SearchService } from 'app/services/search.service';
+import { ApiService } from 'app/services/api';
+import { MatSnackBarRef, SimpleSnackBar, MatSnackBar } from '@angular/material';
+import { CommentPeriodService } from 'app/services/commentperiod.service';
+
+class ProjectNotificationFilterObject {
+  constructor(
+    public type: object = {},
+    public pcp: object = {},
+    public region: Array<string> = []
+  ) { }
+}
+
+@Component({
+  selector: 'app-project-notifications',
+  templateUrl: './project-notifications.component.html',
+  styleUrls: ['./project-notifications.component.scss']
+})
+
+export class ProjectNotificationsListComponent implements OnInit, OnDestroy {
+  private ngUnsubscribe: Subject<boolean> = new Subject<boolean>();
+  public regions: Array<object> = [];
+  public commentPeriods: Array<object> = [];
+  public projectTypes: Array<object> = [];
+  public loading = true;
+  public tableData: TableObject;
+  public tableParams: TableParamsObject = new TableParamsObject();
+  public terms = new SearchTerms();
+  public showAdvancedSearch = true;
+  public filterForURL: object = {};
+  public filterForAPI: object = {};
+  public filterForUI: ProjectNotificationFilterObject = new ProjectNotificationFilterObject();
+
+  public projectNotifications: Array<ProjectNotification> = [];
+  public readonly constants = Constants;
+
+  private snackBarRef: MatSnackBarRef<SimpleSnackBar> = null;
+
+  public showFilters: object = {
+    type: false,
+    region: false,
+    pcp: false
+  };
+
+  public numFilters: object = {
+    type: 0,
+    region: 0,
+    pcp: 0
+  };
+
+  constructor(
+    private _changeDetectionRef: ChangeDetectorRef,
+    private route: ActivatedRoute,
+    private router: Router,
+    private tableTemplateUtils: TableTemplateUtils,
+    private searchService: SearchService,
+    private api: ApiService,
+    public snackBar: MatSnackBar,
+    private commentPeriodService: CommentPeriodService
+  ) {
+
+    this.regions = Constants.REGIONS_COLLECTION;
+    this.commentPeriods = Constants.PCP_COLLECTION;
+    this.projectTypes = Constants.PROJECT_TYPE_COLLECTION;
+  }
+
+  ngOnInit() {
+    this.route.params
+      .takeUntil(this.ngUnsubscribe)
+      .subscribe(params => {
+        this.tableParams = this.tableTemplateUtils.getParamsFromUrl(params);
+        if (this.tableParams.sortBy === '-datePosted') {
+          this.tableParams.sortBy = '-startDate';
+          this.tableTemplateUtils.updateUrl(this.tableParams.sortBy, this.tableParams.currentPage, this.tableParams.pageSize, null, this.tableParams.keywords);
+        }
+        this._changeDetectionRef.detectChanges();
+
+        this.route.data
+          .takeUntil(this.ngUnsubscribe)
+          .subscribe((res: any) => {
+            if (res.projectNotifications[0].data) {
```

Comment:
I get this error here when I dont have any Notification Projects. We should probably have some null checks here, or hide the button or something.

ERROR TypeError: Cannot read property '0' of null
    at SafeSubscriber._next (project-notifications.component.ts:92)